pkg/sdk/views_gen.go

```diff
@@ -0,0 +1,217 @@
+	unset         bool   `ddl:"static" sql:"UNSET"`
+	maskingPolicy bool   `ddl:"static" sql:"MASKING POLICY"`
+}
+
+type ViewSetColumnTags struct {
+	alter   bool             `ddl:"static" sql:"ALTER"`
+	column  bool             `ddl:"static" sql:"COLUMN"`
+	Name    string           `ddl:"keyword"`
+	SetTags []TagAssociation `ddl:"keyword" sql:"SET TAG"`
+}
+
+type ViewUnsetColumnTags struct {
+	alter     bool               `ddl:"static" sql:"ALTER"`
+	column    bool               `ddl:"static" sql:"COLUMN"`
+	Name      string             `ddl:"keyword"`
+	UnsetTags []ObjectIdentifier `ddl:"keyword" sql:"UNSET TAG"`
+}
+
+// DropViewOptions is based on https://docs.snowflake.com/en/sql-reference/sql/drop-view.
+type DropViewOptions struct {
+	drop     bool                   `ddl:"static" sql:"DROP"`
+	view     bool                   `ddl:"static" sql:"VIEW"`
+	IfExists *bool                  `ddl:"keyword" sql:"IF EXISTS"`
+	name     SchemaObjectIdentifier `ddl:"identifier"`
+}
+
+// ShowViewOptions is based on https://docs.snowflake.com/en/sql-reference/sql/show-views.
+type ShowViewOptions struct {
+	show       bool       `ddl:"static" sql:"SHOW"`
+	Terse      *bool      `ddl:"keyword" sql:"TERSE"`
+	views      bool       `ddl:"static" sql:"VIEWS"`
+	Like       *Like      `ddl:"keyword" sql:"LIKE"`
+	In         *In        `ddl:"keyword" sql:"IN"`
+	StartsWith *string    `ddl:"parameter,no_equals,single_quotes" sql:"STARTS WITH"`
+	Limit      *LimitFrom `ddl:"keyword" sql:"LIMIT"`
+}
+
+type viewDBRow struct {
+	CreatedOn      string         `db:"created_on"`
+	Name           string         `db:"name"`
+	Kind           sql.NullString `db:"kind"`
+	Reserved       sql.NullString `db:"reserved"`
+	DatabaseName   string         `db:"database_name"`
+	SchemaName     string         `db:"schema_name"`
+	Owner          sql.NullString `db:"owner"`
+	Comment        sql.NullString `db:"comment"`
+	Text           sql.NullString `db:"text"`
+	IsSecure       sql.NullBool   `db:"is_secure"`
+	IsMaterialized sql.NullBool   `db:"is_materialized"`
+	OwnerRoleType  sql.NullString `db:"owner_role_type"`
+	ChangeTracking sql.NullString `db:"change_tracking"`
+}
+
+type View struct {
+	CreatedOn      string
+	Name           string
+	Kind           string
+	Reserved       string
+	DatabaseName   string
+	SchemaName     string
+	Owner          string
+	Comment        string
+	Text           string
+	IsSecure       bool
+	IsMaterialized bool
+	OwnerRoleType  string
+	ChangeTracking string
+}
+
+func (v *View) ID() SchemaObjectIdentifier {
+	return NewSchemaObjectIdentifier(v.DatabaseName, v.SchemaName, v.Name)
+}
+
+// DescribeViewOptions is based on https://docs.snowflake.com/en/sql-reference/sql/desc-view.
+type DescribeViewOptions struct {
+	describe bool                   `ddl:"static" sql:"DESCRIBE"`
+	view     bool                   `ddl:"static" sql:"VIEW"`
+	name     SchemaObjectIdentifier `ddl:"identifier"`
+}
+
+// TODO: extract common type for describe
```

Comment:
It is explained in the comment below: `viewDetailsRow is a copy of externalTableColumnDetailsRow.`

There are multiple objects which use exactly the same type for response, so it could become our common type, but this is outside the scope of this change and I will add it to our bag of possible improvements of SDK.

Reply:
Oh ok

Reply:
I have added SNOW-965322 to the comment, you can check it out @sfc-gh-jcieslak .